src/test/ui/consts/const_discriminant.rs

```diff
@@ -0,0 +1,22 @@
+// run-pass
+#![feature(const_discriminant)]
+
+use std::mem::{discriminant, Discriminant};
+
+enum Test {
+    A(u8),
+    B,
+    C { a: u8, b: u8 },
+}
```

Comment:
Could you also test an enum like the following, as those are represented somewhat differently?
```rust
enum SingleVariant {
  V,
  Never(!),
}
```

Reply:
done